File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
bool help(string s, string t) {
    if(s.size() != t.size()) return false;
    map<char,char> m;
    for(int i=0; i<s.size(); i++) {
        if(m.find(s[i]) != m.end() && m[s[i]] != t[i]) return false;
        else m[s[i]] = t[i];
    }
    return true;
}

class Solution {
public:
    bool isIsomorphic(string s, string t) {
        return help(s,t) && help(t,s);
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean help(String s,String t) {
        if(s.length() != t.length()) return false;
        HashMap<Character,Character> map = new HashMap<>();
        for(int i=0; i<s.length(); i++) {
            char ch1 = s.charAt(i);
            char ch2 = t.charAt(i);
            if(map.containsKey(ch1) && map.get(ch1) != ch2) return false;
            map.put(ch1,ch2);
        }
        return true;
    }


    public boolean isIsomorphic(String s, String t) {
        return help(s,t) && help(t,s);
    }
}